bpfman/src/serve.rs

```diff
@@ -138,21 +160,18 @@ async fn join_listeners(listeners: Vec<JoinHandle<()>>) {
 async fn serve_unix(
     path: String,
     service: BpfmanServer<BpfmanLoader>,
+    use_activity_timer: &mut bool,
 ) -> anyhow::Result<JoinHandle<()>> {
-    // Listen on Unix socket
-    if Path::new(&path).exists() {
-        // Attempt to remove the socket, since bind fails if it exists
-        remove_file(&path)?;
-    }
-
-    let uds = UnixListener::bind(&path)?;
-    let uds_stream = UnixListenerStream::new(uds);
-    // Always set the file permissions of our listening socket.
-    set_file_permissions(&path.clone(), SOCK_MODE).await;
+    let uds_stream = if let Ok(stream) = systemd_unix_stream(path.clone()) {
+        *use_activity_timer = true;
+        stream
+    } else {
+        std_unix_stream(path.clone()).await?
+    };
```

Comment:
You could just assign a shutdown handler var here here OR just call serve with the different shutdown methods